### ISSUE TITLE
Importing XML case definition file using DSPH

### ISSUE THREAD

**Issue by eleclizard** [DualSPHysics/DesignSPHysics#39]
Hi. 

I want to write my own case definition using the Case_Def.xml file format and then load it with DSPH marco in FreeCAD. I notices there is an "Import XML" option in this current version but when I clicked it it told me that this version doesn't support that feature. Would anyone mind telling me what sort of alternative way that I can import a Case_Def.xml file through DesignSPHysics or maybe do this with the default DualSPHysics files? 

Thank you!

**Comment by zlotny** [comment]
Hi there. The option to Import XML was an idea I had a long time ago because I thought it would be very handy.

However implementing that is very costly and, frankly, I will probably discard it entirely in the future. FreeCAD and DualSPHysics both measure and compute objects in a very different way (units, origin references, parametric vs mesh...) and "understanding" the XML with FreeCAD own way of defining things is hard.

I know it would be a convenient addition but at least for now I'm busy enough with other features in both DualSPHysics and DesignSPHysics so I can't add it.

Sorry about the inconvenience.

To directy answer your question, the only option for now is to redo the case entirely using the DesignSPHysics tools. 

However I never met anyone wanting to write the case in XML to then use DesignsSPHysics with it. Is there any specific reason for that? Just curious :)

**Comment by eleclizard** [comment]
Thank you for your reply!

The reason why want to do this is based on my reading of the case definition file for the DEM example Many_Floating_Def.xml. Upon reading this xml file I found out it is quite inconvenient to go through FreeCAD GUI and define every single solid particle. So I though it could be handy to prepare/further modify a case definition xml file and read it through DesignSPHysics. I am relatively new to SPH method and DualSPHysics and since the tutorials on the website is all about DesignSPHysics so I'm kind of stuck with only knowing how to use DesignSPHysics to set up a simulation. 

If you could link me to a resource that explain the process of defining a case, running gencase, and then run the post processing tool with just DualSPHysics I would really appreciate it! This would make my need to run a xml file through DesignSPHysics redundant it would seem.

**Comment by zlotny** [comment]
I understand learning how to use DualSPHysics in a command line environment is easy, that's why we are making an effort making this GUI toolkit :)

The main resource I can think of is the official wiki: https://github.com/DualSPHysics/DualSPHysics/wiki/5.-Running-DualSPHysics

The entire wiki has a lot of information on what everything does and what different options there are for running the different applications (pre, simulation and post).

To know how to run a case I advise you to just check out the batch files included in each example. In a XXX.bat you'll find how the exact command line for GenCase works (for example) and which options are selected. The same for DualSPHysics, and then each case has it's own post-processing procedures applied, so by investigating a bit the batch file for each example you'll learn a lot.

By the way you wrote this I get that you already understand what's happening on the XML, and that's helpful. The way I learn about the 2 things (XML + Command line) was basically to run an example and look bot to the definition xml and batch file, so I could have an idea of what was happening.

Also take a look at the forum as there's people there that can help you to understand different options and parameters that are not included on examples or not clear enough to learn and understand.

I'm not an expert myself on the DualSPHysics so I'm probably not the right person to answer specific questions, but well I had to learn at least how to use the command line options and XML to make DesignSPHysics, and that's how I did it :)

I hope this helps!
